akka-testkit-typed/src/main/scala/akka/testkit/typed/BehaviourTestkit.scala

```diff
@@ -191,6 +191,10 @@ class BehaviorTestkit[T] private (_name: String, _initialBehavior: Behavior[T])
   def run(msg: T): Unit = {
     try {
       current = Behavior.canonicalize(Behavior.interpretMessage(current, ctx, msg), current, ctx)
+      ctx.executionContext match {
+        case controlled: ControlledExecutor ⇒ controlled.runAll()
```

Comment:
@chbatey It might not be the best choice to use the sync testkit for this stashing example, but I wanted to try it. Then I found that the ControlledExecutor is not doing anything. I was thinking we could at least try to run it here. My example stopped on that the futures callbacks were not run.